### DISCUSSION TITLE
I made an application, it downloads files, but when I install the program files directory, it downloads very slowly, 1 mb in 5 minutes, but you give admin permission, it's okay, what is the reason?

### DISCUSSION THREAD

**Discussion by Meguais** [hgouveia/node-downloader-helper#97] [General]
I made an application, it downloads files, but when I install the program files directory, it downloads very slowly, 1 mb in 5 minutes, but you give admin permission, it's okay, what is the reason?

**Comment by hgouveia** [comment]
@Meguais unfortunately this is not controlled by this library and my best guess is that it might be a OS level thing like QoS

QoS (Quality of Service) is a policy applied in the router or OS to prioritize network bandwidth, in some OSs this gets prioritize when is run as Admin Rights

**Comment by Meguais** [comment]
do i have a chance to prioritize i can run it as admin without running it as admin or at least once ?

**Reply by hgouveia** [reply]
this issues was brought before in https://github.com/hgouveia/node-downloader-helper/issues/47 , and some ppl commented some workarounds 

my best guest is to use `keepAlive = true` and increment the  `maxSockets ` by  creating a new Agent , you are able to pass a custom agent in the settings:

```js
const dl = new DownloaderHelper(url, destFolder, {
  httpsRequestOptions: { agent: new https.Agent({keepAlive: true, maxSockets: 1000 ); }
});
```
you can see more options here and play with them, https://nodejs.org/api/http.html#new-agentoptions , you can reuse the same agent between new DownloaderHelper or you can apply this globally with `http.globalAgent`

but unfortunately this is out of the scope of this library, you will need to research about this on nodejs or electron in case you are using it to go in depth

**Comment by Meguais** [comment]
thank you i solved

**Comment by Meguais** [comment]
Did I run the program in the wrong place, I still can't figure it out 
i tried many things
`
const url = 'https://sabnzbd.org/tests/internetspeed/50MB.bin'; // https://proof.ovh.net/files/
const dl = new DownloaderHelper(url, downloadDir,options, {
  httpsRequestOptions: {agent: new https.Agent({keepAlive: true, maxSockets: 1000, hostname: 'https://sabnzbd.org',})}
});
dl
    .once('download', () => (dl, 5000))
    .on('download', downloadInfo => console.log('Download Begins: ',
        {
            name: downloadInfo.fileName,
            total: downloadInfo.totalSize
        }))
    .on('end', downloadInfo => console.log('Download Completed: ', downloadInfo))
    .on('skip', skipInfo =>
        console.log('Download skipped. File already exists: ', skipInfo))
    .on('error', err => console.error('Something happened', err))
    .on('retry', (attempt, opts, err) => {
        console.log({
            RetryAttempt: `${attempt}/${opts.maxRetries}`,
            StartsOn: `${opts.delay / 1000} secs`,
            Reason: err ? err.message : 'unknown'
        });
    })

    .on('stateChanged', state => console.log('State: ', state))
    .on('renamed', filePaths => console.log('File Renamed to: ', filePaths.fileName))
    .on('redirected', (newUrl, oldUrl) => console.log(`Redirect from '${newUrl}' => '${oldUrl}'`))
    .on('progress', stats => {
    `

**Reply by hgouveia** [reply]
@Meguais  i actually dont know what are you referring to, what issue do you have?  also btw you dont need to specify the `hostname` in the Agent options, in case you need, you need to put it without the `https://` part

**Comment by Meguais** [comment]
I didn't specify the hostname, I tried your code directly, it didn't work again, it's downloading the file very slowly, this is my problem

`const dl = new DownloaderHelper(url, destFolder, {
  httpsRequestOptions: { agent: new https.Agent({keepAlive: true, maxSockets: 1000 ); }
});`
